src/components/ComponentsContext/components/Link/Link.js

```diff
@@ -14,13 +14,29 @@
  */
 
 import React from 'react';
+import isPropValid from '@emotion/is-prop-valid';
 
 import { childrenPropType } from '../../../../util/shared-prop-types';
 
 /**
  * A barebones Link component that's basically just an `<a>` tag
  */
-const Link = ({ children, ...props }) => <a {...props}>{children}</a>;
+const Link = ({ children, ...props }) => (
+  <a {...filterValidProps(props)}>{children}</a>
```

Comment:
Emotion has built-in support for filtered prop forwarding using the [`shouldForwardProp` option](https://emotion.sh/docs/styled#customizing-prop-forwarding). Using this approach should simplify your code quite a bit since you don't need to iterate through the props yourself.

Reply:
Actually, my initial approach was done with the use of `shouldForwardProp`. As I've started adding `shoudForwardProp` to the components which are using `Link`, I've realized that there are 3 of them (`PlainButton`, `RegularButton` and `NavLink`). In the future there could be more. 

So I've decided to encapsulated the filtering logic into the `Link` itself. This looks more of a "once and for all" solution to me, which should not require additional work when adding new components. But I'm also fine with going back to the first approach, using `shouldForwardProp`.

@connor-baer WDYT?

Reply:
I wasn't aware of the `NavLink`. Nevertheless, the props that are passed to the `Link` component can be different in each case (it's pure coincidence that they're the same in these 3 cases). That's why I don't think it makes sense to maintain this list in a central place. Colocating the filter list with each component also ensures that the list is updated when a new prop is added.

PS: The `Link` component will probably be deprecated anyway in the future since it's not the most intuitive approach.

Reply:
Changed the implementation to `shouldForwardProp`